vars/skipStage.groovy

```diff
@@ -523,9 +519,7 @@ boolean call(Map config = [:]) {
                    (skip_stage_pragma('build') &&
                     rpmTestVersion() == '') ||
                    skip_stage_pragma('test') ||
-                   ((env.BRANCH_NAME.startsWith('weekly-testing') ||
-                    env.BRANCH_NAME.startsWith('provider-testing') ||
-                    env.BRANCH_NAME.startsWith('soak-testing')) &&
+                   (env.BRANCH_NAME.matches('^[a-z]+-testing') &&
```

Comment:
And here.

This RE really wants to be DRYed out by being defined in one place and referenced everywhere but I'm not immediately thinking of a good way to do that.

Maybe a new `vars/*.groovy` function that simply returns the RE as a `String` that can then be called from all of these places, at least in `pipeline-lib` groovy functions.